messaging/src/main/java/org/axonframework/eventhandling/scheduling/jobrunr/JobRunrEventScheduler.java

```diff
@@ -0,0 +1,295 @@
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.GenericEventMessage;
+import org.axonframework.eventhandling.scheduling.EventScheduler;
+import org.axonframework.eventhandling.scheduling.ScheduleToken;
+import org.axonframework.eventhandling.scheduling.SchedulingException;
+import org.axonframework.lifecycle.Lifecycle;
+import org.axonframework.lifecycle.Phase;
+import org.axonframework.messaging.MetaData;
+import org.axonframework.messaging.unitofwork.DefaultUnitOfWork;
+import org.axonframework.messaging.unitofwork.UnitOfWork;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.serialization.SimpleSerializedObject;
+import org.jobrunr.scheduling.JobScheduler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+import javax.annotation.Nonnull;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * EventScheduler implementation that delegates scheduling and triggering to a JobRunr JobScheduler.
+ *
+ * @author Gerard Klijs
+ * @since 4.7.0
+ */
+public class JobRunrEventScheduler implements EventScheduler, Lifecycle {
+
+    private static final Logger logger = LoggerFactory.getLogger(JobRunrEventScheduler.class);
+    private final JobScheduler jobScheduler;
+    private final Serializer serializer;
+    private final TransactionManager transactionManager;
+    private final EventBus eventBus;
+
+    /**
+     * Instantiate a {@link JobRunrEventScheduler} based on the fields contained in the
+     * {@link JobRunrEventScheduler.Builder}.
+     * <p>
+     * Will assert that the {@link JobScheduler}, {@link Serializer} and {@link EventBus} are not {@code null}, and will
+     * throw an {@link AxonConfigurationException} if any of them is {@code null}.
+     *
+     * @param builder the {@link Builder} used to instantiate a {@link JobRunrEventScheduler} instance
+     */
+    protected JobRunrEventScheduler(Builder builder) {
+        builder.validate();
+        jobScheduler = builder.jobScheduler;
+        serializer = builder.serializer;
+        transactionManager = builder.transactionManager;
+        eventBus = builder.eventBus;
+    }
+
+    /**
+     * Instantiate a Builder to be able to create a {@link JobRunrEventScheduler}.
+     * <p>
+     * The {@link TransactionManager} is defaulted to a {@link NoTransactionManager}.
+     * <p>
+     * The {@link JobScheduler}, {@link Serializer} and {@link EventBus} are <b>hard requirements</b> and as such should
+     * be provided.
+     *
+     * @return a Builder to be able to create a {@link JobRunrEventScheduler}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    @SuppressWarnings("rawtypes")
+    public ScheduleToken schedule(Instant triggerDateTime, Object event) {
+        UUID jobIdentifier = UUID.randomUUID();
```

Comment:
Same as `JobRunrDeadlineManager` that will give me a `String` not an `UUID`.

Reply:
Hmm, alright, that's a shame. 
But, that also means we have a very good reason why to stick to `UUID`: because it's required by JobRunr.

Knowing that I think we're good here. 
Thanks for the information, @gklijs!